File: mobile/src/services/backend/MaintenanceAgentClient.ts
import { MaintenanceAnalyzeRequest, MaintenanceAnalyzeResponse } from "./contracts";
import { BackendClient } from "./BackendClient";

export class MaintenanceAgentClient {
  constructor(private readonly backend = new BackendClient()) {}

  analyze(request: MaintenanceAnalyzeRequest): Promise<MaintenanceAnalyzeResponse> {
    return this.backend.post<MaintenanceAnalyzeResponse>("/maintenance/analyze", request);
  }
}

File: ACM2/Services/Backend/MaintenanceAgentClient.swift
import Foundation

struct MaintenanceAgentClient {
    private let backend: BackendClient

    init(backend: BackendClient = BackendClient()) {
        self.backend = backend
    }

    func analyze(
        userId: String,
        profile: VehicleProfile,
        reminders: [MaintenanceReminder],
        trips: [TripResult],
        expenses: [MaintenanceExpense]
    ) async throws -> MaintenanceAnalysis {
        try await backend.post(
            "/maintenance/analyze",
            body: MaintenanceAnalyzeRequest(
                userId: userId,
                profile: profile,
                reminders: reminders,
                trips: trips,
                expenses: expenses
            )
        )
    }
}
